### PR TITLE
ENH: add support for NaNs in acovf, acf, pacf (REBASE)

### PR DESCRIPTION
**Pull request by bashtage**
Rebase joesnacks/master which brings missing obs handling to acf

closes #1491

### CODE REVIEW

**Comment by josef-pkt** [comment]
@bashtage  Please put rebase at the end of the title, and keep the correct tag in front ENH, BUG, ... which makes it easier to quickly identify the type of PR.

(I'll go over more PRs after dinner, i.e. in a few hours)

Thanks

File: statsmodels/tsa/tests/test_stattools.py [hunk 1]
```diff
@@ -170,7 +171,47 @@ def test_acf(self):
     def test_qstat(self):
         #todo why is res1/qstat 1 short
         assert_almost_equal(self.res1[1], self.qstat, DECIMAL_3)
-
+        
+class TestACFMissing(CheckCorrGram):
+    """
+    Test Autocorrelation Function using Missing
+    """
```
Comment (josef-pkt) [inline statsmodels/tsa/tests/test_stattools.py]: this should be a code comment `#` not docstring (which nose would use as title)

File: statsmodels/tsa/stattools.py [hunk 2]
```diff
@@ -405,27 +449,10 @@ def acf(x, unbiased=False, nlags=40, qstat=False, fft=False, alpha=None):
...
-        avf = acovf(x, unbiased=unbiased, demean=True)
-        #acf = np.take(avf/avf[0], range(1,nlags+1))
-        acf = avf[:nlags + 1] / avf[0]
-    else:
-        x = np.squeeze(np.asarray(x))
-        #JP: move to acovf
-        x0 = x - x.mean()
-        # ensure that we always use a power of 2 or 3 for zero-padding,
-        # this way we'll ensure O(n log n) runtime of the fft.
-        n = _next_regular(2 * nobs + 1)
```
Comment (josef-pkt) [inline statsmodels/tsa/stattools.py]: before this gets lost: `_next_regular` should be used in acovf
(I don't know why it doesn't show up in the original PR)

Comment (josef-pkt) [inline statsmodels/tsa/stattools.py]: _next_regular merge was after joesnacks branched, PR #1196

File: statsmodels/tsa/tests/test_stattools.py (hunk 1 above)
Comment (bashtage) [inline statsmodels/tsa/tests/test_stattools.py]: I don't think is is true for classes, only tests.  There are many examples of `class TestXXX` that have non-empty docstrings, even in this file (in master).  None-the-less I have changed them.

**Comment by josef-pkt** [comment]
github says merge conflicts, maybe because of warnings merge.

otherwise looks good to me and ready to merge

**Comment by josef-pkt** [comment]
all green on travis, merging

@bashtage  Thank you

**Pull request merged**